File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        L, R = 0, len(height)-1
        maxArea = 0

        while L < R:
            width = R-L
            minHeight = min(height[L], height[R])
            area = width * minHeight
            maxArea = max(maxArea, area)
            if height[L] >= height[R]:
                R-=1
            else:
                L+=1
        return maxArea

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxWater = 0;
        int l = 0;
        int r = height.length-1;
        while (l < r) {
            int area = Math.min(height[l], height[r]) * (r-l);
            maxWater = Math.max(area, maxWater);
            if (height[l] <= height[r]) {
                l++;
            } else {
                r--;
            }
        }
        return maxWater;
    }
}